cli/install/install.go

```diff
@@ -0,0 +1,800 @@
+	flags InstallationFlag, logFile *os.File) error {
+	var err error
+	if checkExisting("tarantool", distfiles) {
+		log.Infof("Local files found, installing from them...")
+		localPath, _ := util.JoinAbspath(distfiles, "tarantool")
+		err = copy.Copy(localPath, path)
+		if err != nil {
+			return err
+		}
+		err = util.ExecuteCommand("git", flags.Verbose, logFile, path, "checkout", tarVersion)
+	} else {
+		return fmt.Errorf("Can't find distfiles directory.")
+	}
+	return err
+}
+
+// copyBuildedTarantool copies binary and include dir.
+func copyBuildedTarantool(binPath, incPath, binDir, includeDir, version string,
+	flags InstallationFlag, logFile *os.File) error {
+	var err error
+	log.Infof("Copying executable...")
+	if _, err := os.Stat(binDir); os.IsNotExist(err) {
+		err = os.Mkdir(binDir, FolderRights)
+		if err != nil {
+			return fmt.Errorf("Unable to create binDir: %s", err)
+		}
+	}
+	if flags.Reinstall {
+		err = os.Remove(filepath.Join(binDir, version))
+		if err != nil {
+			return err
+		}
+	}
+
+	err = util.CopyFilePreverse(binPath, filepath.Join(binDir, version))
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Copying headers...")
+	if _, err := os.Stat(includeDir); os.IsNotExist(err) {
+		err = os.Mkdir(includeDir, 0755)
+		if err != nil {
+			return fmt.Errorf("Unable to create includeDir: %s", err)
+		}
+	}
+	if flags.Reinstall {
+		err = os.Remove(filepath.Join(includeDir, version))
+		if err != nil {
+			return err
+		}
+	}
+	err = copy.Copy(incPath, filepath.Join(includeDir, version)+"/")
+	return err
+}
+
+// installTarantool installs selected version of tarantool.
+func installTarantool(version string, binDir string, incDir string, flags InstallationFlag,
+	distfiles string) error {
+	tarVersion := ""
+	var err error
+	// Get latest tarantool tag if needed.
+	tarVersion = strings.TrimPrefix(version, "tarantool")
+	tarVersion = strings.TrimPrefix(tarVersion, "=")
+	if tarVersion == "" {
+		tarVersion, err = getLatestTarantoolTag()
+		if err != nil {
+			return err
+		}
+		version += "=" + tarVersion
+	}
+	if util.IsDeprecated(tarVersion) {
+		return fmt.Errorf("Deprecated version of tarantool: %s", tarVersion)
+	}
+
+	// Check bin and header dirs.
+	if binDir == "" {
+		return fmt.Errorf("Cannot process bin directory.")
+	}
+	if incDir == "" {
+		return fmt.Errorf("Cannot process include directory.")
+	}
+	logFile, err := ioutil.TempFile("", "tarantool_install")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(logFile.Name())
+
+	log.Infof("Installing tarantool=" + tarVersion)
+
+	// Check if program is already installed.
+	log.Infof("Checking existing...")
+	isDone, err := checkExistingTarantool(version, binDir, flags)
+	if err != nil || isDone {
+		return err
+	}
+
+	// Check dependencies.
+	if !flags.Force {
+		log.Infof("Checking dependencies...")
+		if !programDependenciesInstalled("tarantool") {
+			return nil
+		}
+	}
+
+	path, err := os.MkdirTemp("", "tarantool_install")
+	os.Chmod(path, FolderRights)
```

Comment:
Error check must be done before chmod.
No error check for chmod.

Reply:
changed